packages/vuetify/src/composables/filter.ts

```diff
@@ -0,0 +1,86 @@
+
+// Utilities
+import { getPropertyFromItem, propsFactory, wrapInArray } from '@/util'
+import { computed } from 'vue'
+
+// Types
+import type { PropType, Ref } from 'vue'
+
+export type FilterFunction = typeof defaultFilter
+export type FilterMode = 'intersection' | 'union'
+
+export interface FilterProps {
+  filterFn?: FilterFunction
+  filterMode?: FilterMode
+}
+
+// Composables
+export function defaultFilter (text: string, query?: string) {
```

Comment:
we should probably pass along the entire item, if users need it for whatever reason